internal/exec/exec_test.go

```diff
@@ -0,0 +1,278 @@
+		t.Fatalf("failed to run process: %s", err)
+	}
+}
+
+// Tests that the new procattrlist functionality works as expected.
+func TestExecProcessAttrList(t *testing.T) {
+	procAttrList, procHeap, attrList := createProcAttrList(t, 1)
+	defer winapi.HeapFree(procHeap, 0, attrList)
+	defer winapi.DeleteProcThreadAttributeList(procAttrList)
+
+	// This is useless but it tests all the win32 calls working. A better use for this test would be creating
+	// this process to be a child of a different process other than the one launching it but we can't gurantee
+	// any other processes running/get their pids.
+	pHandle, err := windows.OpenProcess(winapi.PROCESS_ALL_ACCESS, false, uint32(os.Getpid()))
+	if err != nil {
+		t.Fatalf("failed to open process: %s", err)
+	}
+	uintPtrHandle := uintptr(pHandle)
+	if err := winapi.UpdateProcThreadAttribute(
+		procAttrList,
+		0,
+		winapi.PROC_THREAD_ATTRIBUTE_PARENT_PROCESS,
+		&uintPtrHandle,
+		unsafe.Sizeof(pHandle),
+		nil,
+		nil,
+	); err != nil {
+		t.Fatalf("UpdateProcThreadAttribute failed: %s", err)
+	}
+
+	cmd := Command("ping", "127.0.0.1")
+	cmd.SysProcAttr = &SysProcAttr{
+		ProcThreadAttrList: procAttrList,
+	}
+
+	if err := cmd.Run(); err != nil {
+		t.Fatalf("failed to run process: %s", err)
+	}
+}
+
+func TestJobObject(t *testing.T) {
+	procAttrList, procHeap, attrList := createProcAttrList(t, 1)
+	defer winapi.HeapFree(procHeap, 0, attrList)
+	defer winapi.DeleteProcThreadAttributeList(procAttrList)
+
+	job, err := windows.CreateJobObject(nil, windows.StringToUTF16Ptr("test"))
+	if err != nil {
+		t.Fatalf("failed to create job object")
+	}
+	uintPtrHandle := uintptr(job)
+	if err := winapi.UpdateProcThreadAttribute(
+		procAttrList,
+		0,
+		winapi.PROC_THREAD_ATTRIBUTE_JOB_LIST,
+		&uintPtrHandle,
+		unsafe.Sizeof(job),
+		nil,
+		nil,
+	); err != nil {
+		t.Fatalf("UpdateProcThreadAttribute failed: %s", err)
+	}
+
+	cmd := Command("ping", "127.0.0.1")
+	cmd.SysProcAttr = &SysProcAttr{
+		ProcThreadAttrList: procAttrList,
+	}
+
+	if err := cmd.Run(); err != nil {
+		t.Fatalf("failed to run process: %s", err)
+	}
+}
+
+func TestPseudoConsole(t *testing.T) {
+	// Skip if not on RS5. This is when the pseudoconsole API was introduced
+	if osversion.Get().Build < osversion.RS5 {
+		t.Skip()
+	}
+
+	cmd := Command("ping", "127.0.0.1")
+
+	size := &windows.Coord{
+		X: 80,
+		Y: 32,
+	}
+
+	var (
+		ptyIn     windows.Handle
+		ptyOut    windows.Handle
+		pipeFdOut windows.Handle
+		pipeFdIn  windows.Handle
+	)
+
+	if err := windows.CreatePipe(&ptyIn, &pipeFdIn, nil, 0); err != nil {
+		t.Fatalf("failed to create pty input pipes: %s", err)
+	}
+	defer windows.CloseHandle(pipeFdIn)
+
+	if err := windows.CreatePipe(&pipeFdOut, &ptyOut, nil, 0); err != nil {
+		t.Fatalf("failed to create pty output pipes: %s", err)
+	}
+	defer windows.CloseHandle(pipeFdOut)
+
+	coord := uintptr(unsafe.Pointer(size))
+	var hpCon windows.Handle
+	if err := winapi.CreatePseudoConsole(
+		coord,
```

Comment:
`CreatePseudoConsole` doesn't take a `*COORD`, so I don't see how this can be passing the correct value here. :)

I think we will need to pack the correct value into a `uintptr` and pass it by value.